service/history/shard/controller.go

```diff
@@ -91,6 +96,7 @@ type (
 
 		sync.RWMutex
 		historyShards map[int]*historyShardsItem
+		snapshot      atomic.Pointer[shardIDSnapshot]
```

Comment:
nit: more specific naming
```suggestion
		shardIDSnapshot      atomic.Pointer[shardIDSnapshot]
```